File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/user/UserCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.user;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class UserCreator {

    @Builder
    public static User create(
        @NotNull String socialId,
        @NotNull UserSocialType socialType,
        @NotNull String name
    ) {
        return User.builder()
            .socialId(socialId)
            .socialType(socialType)
            .name(name)
            .build();
    }

}

File: threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/user/UserCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.user

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object UserCreator {

    @JvmStatic
    fun create(
            socialId: String,
            socialType: UserSocialType,
            name: String
    ): User {
        return User.builder()
            .socialId(socialId)
            .socialType(socialType)
            .name(name)
            .build()
    }

}
